### PR TITLE
Run typedoc only when entrypoints were changed

### PR DESCRIPTION
**Pull request by bragaru-i**
Run `typedoc` only when file are changed:
- get `entryPoints` from [typedoc.json](https://github.com/Collaborne/video-player/blob/27d71bdfce17d25c62214ea71d12d6ec42c919ee/typedoc.json#L7)
- get all files that are included in `entryPoints`
- check their relative path with git diffed value from `origin/main`

### CODE REVIEW

File: tasks/run-typedoc-on-updates.ts [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+
+const committedFilePaths = executeGitCommand('git diff --name-only main');
+
+const committedRelativeFilePaths = resolveFilePaths(committedFilePaths);
+
+const TYPEDOC_DIR = path.resolve(__dirname, '../typedoc.json');
+
+const typeDocInput: string[] = readFile(TYPEDOC_DIR)?.entryPoints ?? [];
+
+const getDocumentedFiles = () => {
```
Comment (ronnyroeller) [inline tasks/run-typedoc-on-updates.ts:37]: I think you can simplify to:
```ts
const documentedFiles = typeDocInput.flatMap(
  entryPoint => glob(entryPoint, { realpath: true },
));
```

Comment (ronnyroeller) [reply tasks/run-typedoc-on-updates.ts:37]: What does `documentedFiles` mean? Files that contain the documentation or files that are documented?

File: tasks/run-typedoc-on-updates.ts [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+		entryRelativeFiles.push(...files);
+	});
+
+	return entryRelativeFiles;
+};
+
+const documentedFiles = getDocumentedFiles();
+
+let shouldRunDocUpdate = false;
+const fileUpdated: string[] = [];
```
Comment (ronnyroeller) [inline tasks/run-typedoc-on-updates.ts:50]: Simplify to:
```ts
const fileUpdated = documentedFiles.filter(file => committedRelativeFilePaths.includes(file));
const shouldRunDocUpdate = fileUpdated.length > 0;
```

Comment (ronnyroeller) [reply tasks/run-typedoc-on-updates.ts:50]: `fileUpdated` is unclear. Does it mean `updatedFiles` (or `changedFiles`)?

**Comment by bragaru-i** [comment]
@eacet , need your help here.
[build](https://github.com/Collaborne/video-player/actions/runs/3331410635/jobs/5511113508) fails because main is not pulled? Am I right?

File: tasks/run-typedoc-on-updates.ts [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+});
+
+if (shouldRunDocUpdate) {
+	console.group('Creating new docs due to changes in');
+	console.log(fileUpdated);
+	execSync('typedoc');
+
+	process.exit(0);
+}
+console.log('No pending updates for Typedoc ');
```
Comment (ronnyroeller) [inline tasks/run-typedoc-on-updates.ts:66]: Better add `else`. although not strictly required, it makes it easier to read.
```ts
if (...) {
} else {
  console.log('No pending updates for Typedoc ');
}
```

Comment (ronnyroeller) [inline tasks/run-typedoc-on-updates.ts:66]: Remove unnecessary final space

File: tasks/run-typedoc-on-updates.ts [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+
+documentedFiles.forEach(file => {
+	if (committedRelativeFilePaths.includes(file)) {
+		shouldRunDocUpdate = true;
+		fileUpdated.push(file);
+	}
+});
+
+if (shouldRunDocUpdate) {
+	console.group('Creating new docs due to changes in');
```
Comment (ronnyroeller) [inline tasks/run-typedoc-on-updates.ts:60]: Improve wording: "Recreating docs due to changes in"

**Review by ronnyroeller — APPROVED** [review]
LGTM (pls process comments before merging)

File: tasks/run-typedoc-on-updates.ts (hunk 1 above)
Comment (bragaru-i) [reply tasks/run-typedoc-on-updates.ts:37]: entryPoint files for `typedoc` (that matches the pattern)

Comment (ronnyroeller) [reply tasks/run-typedoc-on-updates.ts:37]: > entryPoint files for typedoc (that matches the pattern)

Sorry, what I meant: could you adjust the variable to communicate clearer the purpose of it.

**Comment by bragaru-i** [comment]
Closing PR due to its non-relevance anymore.
Now documentation deploys assets and deploys them on every PR into main branch (same as Storybook)

**Pull request closed without merging**